### PR TITLE
✨ Refactor ReaGroup GenomicFile Relation

### PR DESCRIPTION
**Pull request by znatty22**
This PR changes the 1 to 1 relationship between ReadGroup and GenomicFile to be many to many. The `paired_end` field, previously on ReadGroup will now be on the GenomicFile.

Also, orphaned read groups (those which do not have any associated genomic files) are automatically deleted when they become orphans

### CODE REVIEW

File: config.py [hunk 1]
```diff
@@ -44,7 +44,7 @@ class TestingConfig(Config):
     SERVER_NAME = "localhost"
     TESTING = True
     WTF_CSRF_ENABLED = False
-    # SQLALCHEMY_DATABASE_URI = 'postgres://postgres@localhost:5432/test'
+    SQLALCHEMY_DATABASE_URI = 'postgres://postgres@localhost:5432/test'
```
Comment (dankolbman) [inline config.py]: This is killing Jenkins

File: dataservice/api/read_group/models.py [hunk 2]
```diff
@@ -17,17 +20,47 @@ class ReadGroup(db.Model, Base):
...
+                              nullable=False)
+
+    genomic_file_id = db.Column(KfId(),
+                                db.ForeignKey('genomic_file.kf_id'),
                                 nullable=False)
+    read_group = db.relationship('ReadGroup')
+    genomic_file = db.relationship('GenomicFile')
+
+    def __repr__(self):
+        return "{}-{}".format(self.read_group.external_id,
```
Comment (dankolbman) [inline dataservice/api/read_group/models.py]: Do we use this format anywhere else?
We should probably stay close to the typical `<Object >` format that python uses.
Maybe `<ReadGroupGenomicFile {self.read_group.__repr__()} - {self.genomic_file.__repr__()}>`?

Comment (znatty22) [reply dataservice/api/read_group/models.py]: Ah sorry, that was there for debugging, let me change it

File: dataservice/api/genomic_file/resources.py [hunk 3]
```diff
@@ -58,6 +61,11 @@ def get(self, filter_params, after, limit):
                  .join(Biospecimen.participant)
                  .filter(Participant.study_id == study_id))
 
+        from dataservice.api.read_group.models import ReadGroupGenomicFile
+        if read_group_id:
+            q = (q.join(ReadGroupGenomicFile)
+                 .filter(ReadGroupGenomicFile.read_group_id == read_group_id))
```
Comment (parimalak) [inline dataservice/api/genomic_file/resources.py:67]: I think this query still have the same problem what we have with biospecimen-diagnosis

File: dataservice/api/read_group/models.py [hunk 4]
```diff
@@ -17,17 +20,43 @@ class ReadGroup(db.Model, Base):
...
+                                          'genomic_file_id'),)
+    read_group_id = db.Column(KfId(),
+                              db.ForeignKey('read_group.kf_id'),
+                              nullable=False)
+
+    genomic_file_id = db.Column(KfId(),
+                                db.ForeignKey('genomic_file.kf_id'),
                                 nullable=False)
+    read_group = db.relationship('ReadGroup')
+    genomic_file = db.relationship('GenomicFile')
```
Comment (parimalak) [inline dataservice/api/read_group/models.py:55]: are these relationships used in the delete listening event?

File: dataservice/api/genomic_file/resources.py (hunk 3 above)
Comment (znatty22) [reply dataservice/api/genomic_file/resources.py:67]: It doesn't, I tested it to make sure we're not getting back extra genomic files

File: dataservice/api/read_group/models.py (hunk 4 above)
Comment (znatty22) [reply dataservice/api/read_group/models.py:55]: No they're not used, but they're used in tests. Also just FYI, the delete listening event only deletes read groups that have no genomic files associated with them. There is no automatic deletion of genomic files

**Review by dankolbman — APPROVED** [review]
👍

**Review by parimalak — APPROVED** [review]
update erd before merging.

**Pull request merged**